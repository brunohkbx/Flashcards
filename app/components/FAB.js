import React from 'react';
import PropTypes from 'prop-types';
import { FAB as _FAB, withTheme } from 'react-native-paper';

export const FAB = ({ primary, handlePress, theme }) => {
  const { colors } = theme;
  const backgroundColor = primary ? colors.primary : colors.accent;

  return (
    <_FAB
      dark
      style={{backgroundColor}}
      icon="add"
      onPress={handlePress}
      style={{margin: 10}}
    />
  );
};

FAB.propTypes = {
  primary: PropTypes.bool,
  handlePress: PropTypes.func.isRequired
}

FAB.defaultProps = {
  primary: false
}

export default withTheme(FAB);
