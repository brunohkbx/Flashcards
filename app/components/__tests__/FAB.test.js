import React from 'react';
import { shallow } from 'enzyme';
import { DefaultTheme } from 'react-native-paper';
import { FAB } from '../FAB';

describe('FAB', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      handlePress: jest.fn(),
      theme: DefaultTheme
    }, propOverrides);

    const wrapper = shallow(<FAB {...defaultProps} />);

    const press = () => {
      wrapper.simulate('press');
    };

    return {
      defaultProps,
      wrapper,
      press
    };
  };

  it('changes background-color to primary when primary is true', () => {
    const { wrapper } = setup({ primary: true });

    expect(wrapper).toMatchSnapshot();
  });

  it('changes background-color to secondary when primary is false', () => {
    const { wrapper } = setup({ primary: false });

    expect(wrapper).toMatchSnapshot();
  });

  test('onPress calls handlePress', () => {
    const mockHandlePress = jest.fn();
    const { press } = setup({ handlePress: mockHandlePress });

    press();

    expect(mockHandlePress).toHaveBeenCalled();
  });
});
