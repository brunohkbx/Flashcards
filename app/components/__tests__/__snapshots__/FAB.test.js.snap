// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAB changes background-color to primary when primary is true 1`] = `
<withTheme(FAB)
  dark={true}
  icon="add"
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#3f51b5",
    }
  }
/>
`;

exports[`FAB changes background-color to secondary when primary is false 1`] = `
<withTheme(FAB)
  dark={true}
  icon="add"
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#ff4081",
    }
  }
/>
`;
